### ISSUE TITLE
Build failure on Mac

### ISSUE THREAD

**Issue by cdunklau** [Haivision/srt#5]
I installed cmake and openssl with brew, and got this build failure. I upgraded everything with brew, still got it.

Mac version is 10.12.4 (16E195)

I do not have XCode installed, only the command line tools (installed with `xcode-select --install`, so maybe that has something to do with it.

**Brew info for cmake and openssl**

```
hobbes:srt cdunklau$ brew list cmake openssl
/usr/local/Cellar/cmake/3.8.0/bin/ccmake
/usr/local/Cellar/cmake/3.8.0/bin/cmake
/usr/local/Cellar/cmake/3.8.0/bin/cmakexbuild
/usr/local/Cellar/cmake/3.8.0/bin/cpack
/usr/local/Cellar/cmake/3.8.0/bin/ctest
/usr/local/Cellar/cmake/3.8.0/share/aclocal/cmake.m4
/usr/local/Cellar/cmake/3.8.0/share/cmake/ (2170 files)
/usr/local/Cellar/cmake/3.8.0/share/doc/ (8 files)
/usr/local/Cellar/cmake/3.8.0/share/emacs/site-lisp/cmake/cmake-mode.el
/usr/local/Cellar/cmake/3.8.0/share/man/ (19 files)
/usr/local/Cellar/openssl/1.0.2k/bin/c_rehash
/usr/local/Cellar/openssl/1.0.2k/bin/openssl
/usr/local/Cellar/openssl/1.0.2k/include/openssl/ (75 files)
/usr/local/Cellar/openssl/1.0.2k/lib/libcrypto.1.0.0.dylib
/usr/local/Cellar/openssl/1.0.2k/lib/libssl.1.0.0.dylib
/usr/local/Cellar/openssl/1.0.2k/lib/engines/ (12 files)
/usr/local/Cellar/openssl/1.0.2k/lib/pkgconfig/ (3 files)
/usr/local/Cellar/openssl/1.0.2k/lib/ (4 other files)
/usr/local/Cellar/openssl/1.0.2k/share/man/ (1592 files)
```

**OpenSSL that brew installed is linked right**

```
hobbes:~ cdunklau$ ls -ld /usr/local/opt/openssl
lrwxr-xr-x  1 cdunklau  admin  24 Feb  5 17:32 /usr/local/opt/openssl -> ../Cellar/openssl/1.0.2k
```

**Configure and make output:**

```
hobbes:srt cdunklau$ ./configure 
Running: cmake . -DWITH_OPENSSL_INCLUDEDIR=/usr/local/opt/openssl/include -DWITH_OPENSSL_LDFLAGS=/usr/local/opt/openssl/lib/libcrypto.a
-- The C compiler identification is AppleClang 8.1.0.8020038
-- The CXX compiler identification is AppleClang 8.1.0.8020038
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2") 
COMPILER: GNU compat: /Library/Developer/CommandLineTools/usr/bin/c++
VERSION: major=1 minor=2 patch=0
Using RELEASE mode
DARWIN detected
C++ VERSION: Setting C++11 compat flag for gnu compiler
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/cdunklau/Development/srt
hobbes:srt cdunklau$ make
Scanning dependencies of target haicrypt
[  2%] Building C object CMakeFiles/haicrypt.dir/haicrypt/hc_openssl_aes.c.o
[  5%] Building C object CMakeFiles/haicrypt.dir/haicrypt/hc_openssl_evp_cbc.c.o
[  8%] Building C object CMakeFiles/haicrypt.dir/haicrypt/hc_openssl_evp_ctr.c.o
[ 11%] Building C object CMakeFiles/haicrypt.dir/haicrypt/hcrypt.c.o
[ 14%] Building C object CMakeFiles/haicrypt.dir/haicrypt/hcrypt_ctx_rx.c.o
[ 17%] Building C object CMakeFiles/haicrypt.dir/haicrypt/hcrypt_ctx_tx.c.o
[ 20%] Building C object CMakeFiles/haicrypt.dir/haicrypt/hcrypt_rx.c.o
[ 22%] Building C object CMakeFiles/haicrypt.dir/haicrypt/hcrypt_sa.c.o
[ 25%] Building C object CMakeFiles/haicrypt.dir/haicrypt/hcrypt_tx.c.o
[ 28%] Building C object CMakeFiles/haicrypt.dir/haicrypt/hcrypt_ut.c.o
[ 31%] Building C object CMakeFiles/haicrypt.dir/haicrypt/hcrypt_xpt_srt.c.o
[ 34%] Building C object CMakeFiles/haicrypt.dir/haicrypt/hcrypt_xpt_sta.c.o
[ 37%] Linking C static library libhaicrypt.a
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libhaicrypt.a(hc_openssl_evp_cbc.c.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libhaicrypt.a(hc_openssl_evp_cbc.c.o) has no symbols
[ 37%] Built target haicrypt
Scanning dependencies of target haisrt
[ 40%] Building CXX object CMakeFiles/haisrt.dir/srtcore/api.cpp.o
/Users/cdunklau/Development/srt/srtcore/api.cpp:2638:20: warning: unused function 'set_result'
      [-Wunused-function]
static inline void set_result(set<UDTSOCKET>* val, int* num, UDTSOCKET* fds, set<UDTSOCKET>::co...
                   ^
1 warning generated.
[ 42%] Building CXX object CMakeFiles/haisrt.dir/srtcore/buffer.cpp.o
In file included from /Users/cdunklau/Development/srt/srtcore/buffer.cpp:65:
In file included from /Users/cdunklau/Development/srt/srtcore/buffer.h:68:
In file included from /Users/cdunklau/Development/srt/srtcore/list.h:68:
In file included from /Users/cdunklau/Development/srt/srtcore/common.h:77:
/Users/cdunklau/Development/srt/srtcore/utilities.h:582:17: warning: absolute value function 'abs'
      given an argument of type 'int64_t' (aka 'long long') but has parameter of type 'int' which may
      cause truncation of value [-Wabsolute-value]
            if (abs(m_qDrift) > MAX_DRIFT)
                ^
/Users/cdunklau/Development/srt/srtcore/buffer.cpp:1454:34: note: in instantiation of member function
      'DriftTracer<1000, 5000, true>::update' requested here
    bool updated = m_DriftTracer.update(iDrift);
                                 ^
/Users/cdunklau/Development/srt/srtcore/utilities.h:582:17: note: use function 'std::abs' instead
            if (abs(m_qDrift) > MAX_DRIFT)
                ^~~
                std::abs
1 warning generated.
[ 45%] Building CXX object CMakeFiles/haisrt.dir/srtcore/cache.cpp.o
[ 48%] Building CXX object CMakeFiles/haisrt.dir/srtcore/ccc.cpp.o
[ 51%] Building CXX object CMakeFiles/haisrt.dir/srtcore/channel.cpp.o
In file included from /Users/cdunklau/Development/srt/srtcore/channel.cpp:85:
/Users/cdunklau/Development/srt/include/srt_compat.h:88:8: warning: comparison of function
      'clock_gettime' not equal to a null pointer is always true [-Wtautological-pointer-compare]
   if (clock_gettime != NULL)
       ^~~~~~~~~~~~~    ~~~~
/Users/cdunklau/Development/srt/include/srt_compat.h:88:8: note: prefix with the address-of operator to
      silence this warning
   if (clock_gettime != NULL)
       ^
       &
/Users/cdunklau/Development/srt/include/srt_compat.h:101:8: warning: comparison of function
      'clock_getres' not equal to a null pointer is always true [-Wtautological-pointer-compare]
   if (clock_getres != NULL)
       ^~~~~~~~~~~~    ~~~~
/Users/cdunklau/Development/srt/include/srt_compat.h:101:8: note: prefix with the address-of operator
      to silence this warning
   if (clock_getres != NULL)
       ^
       &
/Users/cdunklau/Development/srt/include/srt_compat.h:129:17: error: cannot initialize a variable of
      type 'const char *' with an rvalue of type 'void *'
    const char* pos = memchr(s, 0, maxlen);
                ^     ~~~~~~~~~~~~~~~~~~~~
/Users/cdunklau/Development/srt/include/srt_compat.h:134:5: error: "FIXME!!"
   #error "FIXME!!"
    ^
2 warnings and 2 errors generated.
make[2]: *** [CMakeFiles/haisrt.dir/srtcore/channel.cpp.o] Error 1
make[1]: *** [CMakeFiles/haisrt.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by rndi** [comment]
Thanks Colin. Could you see if #6 fixes the problem on your side?

**Comment by rndi** [comment]
Confirmed by @adamyellen as resolved with #6

**Comment by cdunklau** [comment]
Since #6 was merged, I don't see this problem either.
